org.eclipse.xtext/src/org/eclipse/xtext/workspace/WorkspaceConfig.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.workspace;
+
+import static java.util.Arrays.*;
+import static org.eclipse.xtext.xbase.lib.IterableExtensions.*;
+
+import java.util.HashMap;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+
+/**
+ * The default workspace consists of multiple projects and always contains the unknown project that is used for
+ * projectless files.
+ *
+ * @author kosyakov - Initial contribution and API
+ */
+public class WorkspaceConfig implements IWorkspaceConfig {
+	private final UnknownProjectConfig unknown = new UnknownProjectConfig(this);
+
+	private final HashMap<String, IProjectConfig> projects = new HashMap<>();
+
+	public WorkspaceConfig(final IProjectConfig... initialProjects) {
+		addProject(this.unknown);
+		stream(initialProjects).forEach(p -> addProject(p));
+	}
+
+	@Override
+	public Set<? extends IProjectConfig> getProjects() {
+		return toSet(this.projects.values());
+	}
+
+	public void addProject(final IProjectConfig project) {
+		this.projects.put(project.getName(), project);
+	}
+
+	@Override
+	public IProjectConfig findProjectByName(final String name) {
+		return this.projects.get(name);
+	}
+
+	@Override
+	public IProjectConfig findProjectContaining(final URI member) {
+		return this.projects.values().stream()
```

Comment:
projects is a hashmap which are usually bad for iteration. Should we consider using a LinkedHashMap?

Reply:
ok, done